crates/core/src/transport.rs

```diff
@@ -40,9 +40,13 @@ mod connection_handler;
 mod crypto;
 mod packet_data;
 mod peer_connection;
-mod receipt_tracker;
+mod received_packet_tracker;
+mod sent_packet_tracker;
 mod symmetric_message;
 
+type MessagePayload = Vec<u8>;
+type MessageId = u32;
```

Comment:
so I don't think using u32 is the solution (although running out of u32 probably won't happen), we should just wrap and start counting again from zero, we just need to track for u16::max to know we just wrapped around

this can wait and maing it u32 is simplier but is not necessary

Reply:
The difficulty is that if we're sending 1k per packet then we only need to transmit 65MB before a u16 will wrap around and start reusing u16s. On a 1gb connection that will only take 0.5 seconds which is way too close for comfort IMHO.